Source/Core/Common/StringUtil.cpp

```diff
@@ -197,26 +197,42 @@ std::string StringFromFormatV(const char* format, va_list args)
   return temp;
 }
 
-// For Debugging. Read out an u8 array.
-std::string ArrayToString(const u8* data, u32 size, int line_len, bool spaces)
+// This assumes you already masked the upper four bits.
+constexpr char NibbleToHex(const u8 n)
 {
-  std::ostringstream oss;
-  oss << std::setfill('0') << std::hex;
+  if (n < 0xa)
+    return '0' + n;
+  else
+    return 'A' + n - 0xa;
+}
 
-  for (int line = 0; size; ++data, --size)
+std::string MemToHexString(const void* data, std::size_t size, const std::size_t per_line,
+                           const bool spaces)
+{
+  std::string hexstring;
+  if (size == 0 || per_line == 0)
+    return hexstring;
```

Comment:
```suggestion
    return {};
```

Then just move the actual container down by the reserve call

Reply:
As it is now produces better assembly.